### PR TITLE
Transaction executions recovery

### LINKED ISSUE #255: Fix Contract Redeployment
**Describe the bug**
If a contract is deployed which places a resource onto an account, redeploying this contract will fail. Currently, a transaction will delete the contract. However, this does not delete the transaction which places the resource onto the account. Redeploying the contract will now fail due to the resource already being on the account. This causes projects to not load due to being in this invalid state.

Issue on Sentry: https://dapperlabs.sentry.io/issues/4083790981/?project=6398442&query=is%3Aunresolved&referrer=issue-stream&stream_index=0

- Flow-CLI has the same issue: https://github.com/onflow/flow-cli/issues/986

### Solution
Create a mapping of deployed Contracts to their corresponding block height. When a contract is redeployed, we then reset the block height to before the contract was last deployed. This clears all actions after the initial deployment of the contract which solves the issue of having invalid resources left on accounts due to removing or editing a contract.

We will need to delete all transaction executions + contract deployments in the database past after and including this block height/ index, and then deploy the contract again.

In the database modify ContractDeployments to include block height:

```
ContractDeployment {
	File
	Address     Address        `gorm:"serializer:json"`
	BlockHeight int            `json:"blockHeight"`
	Errors      []ProgramError `gorm:"serializer:json"`
	Events      []Event        `gorm:"serializer:json"`
	Logs        []string       `gorm:"serializer:json"`
}
```

Now instead of resetting the state completely when a user updates a contract, we just rollback to the previous state before their contract was deployed initially.

### PR DESCRIPTION
**Pull request by DylanTinianov**
Closes: #255 

## Description
Recover from Cadence runtime panic which maybe occur when  running transaction executions

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/onflow/flow-playground/blob/master/CONTRIBUTING.md))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [standards mentioned here](https://github.com/onflow/flow-playground/blob/master/CONTRIBUTING.md).
- [ ] Updated relevant documentation 
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Added appropriate labels

### CODE REVIEW

File: blockchain/emulator.go [hunk 1]
```diff
@@ -94,8 +94,14 @@ func (e *emulator) executeTransaction(
 	script string,
 	arguments []string,
 	authorizers []flowsdk.Address,
-) (*types.TransactionResult, *flowsdk.Transaction, error) {
-	tx := &flowsdk.Transaction{}
+) (result *types.TransactionResult, tx *flowsdk.Transaction, err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			err = errors.New("Panic occurred during transaction execution")
```
Comment (devbugging) [inline blockchain/emulator.go:100]: Why is this needed? It hides the details of the panic and panic shouldn't be treated as the error. So I wonder what is your thinking?

Comment (DylanTinianov) [reply blockchain/emulator.go:100]: We actually don't need this 👀

Comment (devbugging) [reply blockchain/emulator.go:100]: Okay :D
